Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/Payment Flow/PaymentPriceViewController.m

```diff
@@ -49,5 +55,11 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
     }
 }
 
+- (BOOL)textFieldShouldEndEditing:(UITextField *)textField {
+    if (textField.text.length == 5) { //FIXME: MUST STOP EDITING AFTER 2 DECIMAL POINTS
```

Comment:
You can use the [`textField:shouldChangeCharactersIn:replacementString:`](https://developer.apple.com/documentation/uikit/uitextfielddelegate/1619599-textfield) delegate method to make sure text input is always a valid number with two decimals. There's probably some third party library you can use for this too.

Reply:
Something like [CurrencyTextField](https://github.com/richa008/CurrencyTextField) might work if implementing yourself becomes too complex. There might be better ones too, or more used (with more starts on GitHub).

Reply:
Stripe might have an UI component for this too.